xmppserver/src/main/java/org/jivesoftware/openfire/disco/DiscoInfoProvider.java

```diff
@@ -67,9 +69,25 @@
      * @param node the requested disco node.
      * @param senderJID the XMPPAddress of user that sent the disco info request.
      * @return an XDataForm with the extended information about the entity or null if none.
+     * @deprecated Replaced by {@link #getExtendedInfos(String, String, JID)}
      */
+    @Deprecated
     DataForm getExtendedInfo( String name, String node, JID senderJID );
 
+    /**
```

Comment:
The Javadoc (which is the "contract" for this method" defines that null is returned when there is no extended information dataform.

This does not correspond with the default implementation: when no form is available, I think the implementation returns a Collection that contains a single, null, element.

Two things:
- I think it's better to return an empty collection, instead of a collection that contains a null value. If a collection with a null value is needed, the reason for this must be clearly documented.
- In any case, the javadoc and the implementation must define the same implementation.